File: app/src/main/java/com/kkkkkn/readbooks/util/eventBus/events/MainEvent.java
package com.kkkkkn.readbooks.util.eventBus.events;

import com.kkkkkn.readbooks.model.entity.BookInfo;
import com.kkkkkn.readbooks.util.eventBus.EventMessage;

import java.util.ArrayList;

public class MainEvent {
    public String token;
    public int accountId;
    public EventMessage message;
    public String url;
    public String path;
    public String name;
    public ArrayList<BookInfo> bookShelf;

    public MainEvent(EventMessage message, ArrayList<BookInfo> bookInfos) {
        this.bookShelf = bookInfos;
        this.message = message;
    }

    public MainEvent(EventMessage message, int accountId,String token) {
        this.token = token;
        this.accountId = accountId;
        this.message = message;
    }

    public MainEvent(EventMessage message,String name,String path,String url, int accountId,String token) {
        this.token = token;
        this.accountId = accountId;
        this.message = message;
        this.url=url;
        this.path=path;
        this.name=name;
    }
}

File: app/src/main/java/com/kkkkkn/readbooks/util/eventBus/events/MainEvent.kt
package com.kkkkkn.readbooks.util.eventBus.events

import com.kkkkkn.readbooks.model.entity.BookInfo
import com.kkkkkn.readbooks.util.eventBus.EventMessage

class MainEvent {
    var token: String? = null
    var accountId = 0
    var message: EventMessage
    var url: String? = null
    var path: String? = null
    var name: String? = null
    var bookShelf: ArrayList<BookInfo>? = null

    constructor(message: EventMessage, bookInfos: ArrayList<BookInfo>?) {
        bookShelf = bookInfos
        this.message = message
    }

    constructor(message: EventMessage, accountId: Int, token: String?) {
        this.token = token
        this.accountId = accountId
        this.message = message
    }

    constructor(
        message: EventMessage,
        name: String?,
        path: String?,
        url: String?,
        accountId: Int,
        token: String?
    ) {
        this.token = token
        this.accountId = accountId
        this.message = message
        this.url = url
        this.path = path
        this.name = name
    }
}